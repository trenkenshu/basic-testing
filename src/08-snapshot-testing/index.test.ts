// Uncomment the code below and write your tests
import { generateLinkedList } from './index';

describe('generateLinkedList', () => {
  // Check match by expect(...).toStrictEqual(...)
  test('should generate linked list from values 1', () => {
    const nums = [1, 2, 3, 4];
    const list1 = generateLinkedList(nums);
    expect(list1).toStrictEqual(generateLinkedList(nums));
  });

  // Check match by comparison with snapshot
  test('should generate linked list from values 2', () => {
    const nums = [1, 2, 3, 4];
    const list1 = JSON.stringify(generateLinkedList(nums));
    expect(list1).toMatchSnapshot();
  });
});
