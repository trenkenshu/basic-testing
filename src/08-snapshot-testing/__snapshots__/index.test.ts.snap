// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateLinkedList should generate linked list from values 2 1`] = `"{"value":1,"next":{"value":2,"next":{"value":3,"next":{"value":4,"next":{"value":null,"next":null}}}}}"`;
